### PR TITLE
Wait for resolve

### PR DESCRIPTION
**Pull request by shans**

### CODE REVIEW

File: core/scheduler.js [hunk 1]
```diff
@@ -18,18 +18,38 @@ var taskQueue = [];
...
     return this.dependencies.length + this.executingDependencies > 0;
+  },
+  waitFor: function(task) {
+    this.waitingFor = task.waitingFor;
+    task.waitingFor = [];
+    this.waitingFor.push(task);
+  },
+  resolveTask: function() {
+    this.finished = true;
+    this.waitingFor = this.waitingFor.filter(function(task) { return task.finished == false; });
```
Comment (dstockwell) [inline core/scheduler.js]: `return !task.finished`

File: core/scheduler.js [hunk 2]
```diff
@@ -18,18 +18,38 @@ var taskQueue = [];
...
+    if (this.waitingFor.length == 0) {
+      this.resolve();
+      return;
+    }
+    var newWaitingTask = this.waitingFor.pop();
```
Comment (dstockwell) [inline core/scheduler.js]: might warrant a comment, confusing without your earlier explanation today

**Comment by dstockwell** [comment]
lgtm

File: core/scheduler.js (hunk 1 above)
Comment (shans) [inline core/scheduler.js]: done

File: core/scheduler.js (hunk 2 above)
Comment (shans) [inline core/scheduler.js]: done

**Pull request merged**